bokehjs/src/coffee/models/glyphs/bokehgl.coffee

```diff
@@ -48,30 +48,34 @@ attach_float = (prog, vbo, att_name, n, visual, name) ->
       vbo.set_size(n*4)
       vbo.set_data(0, a)
       prog.set_attribute(att_name, 'float', [vbo, 0, 0])
-    return a
 
-attach_color = (prog, vbo, att_name, n, visual) ->
+attach_color = (prog, vbo, att_name, n, visual, prefix) ->
     # Attach the color attribute to the program. If there's just one color,
     # then use this single color for all vertices (no VBO). Otherwise we
     # create an array and upload that to the VBO, which we attahce to the prog.
     m = 4
     vbo.used = true
-    if visual.color.spec.value? and visual.alpha.spec.value?
-      rgba = color2rgba(visual.color.spec.value, visual.alpha.spec.value)
+    colorname = prefix + '_color'
+    alphaname = prefix + '_alpha'
+    if visual[colorname].spec.value? and visual[alphaname].spec.value?
+      rgba = color2rgba(visual[colorname].spec.value, visual[alphaname].spec.value)
       prog.set_attribute(att_name, 'vec4', null, rgba)
       vbo.used = false
 
     else
       # Get array of colors
-      if visual.color.spec.value?
-        colors = (visual.color.spec.value for i in [0...n])
-      else
-        colors = visual.cache.color_array
+      if visual[colorname].spec.value?
+        colors = (visual[colorname].spec.value for i in [0...n])
+      else if visual.cache[colorname+'_array']?
+        colors = visual.cache[colorname+'_array']
+      else  # color is None / null
+        prog.set_attribute(att_name, 'vec4', null, [0,0,0,0])
+        return
       # Get array of alphas
-      if visual.alpha.spec.value?
-        alphas = fill_array_with_float(n, visual.alpha.spec.value)
+      if visual[alphaname].spec.value?
+        alphas = fill_array_with_float(n, visual[alphaname].spec.value)
       else
-        alphas = visual.cache.alpha_array
+        alphas = visual.cache[alphaname+'_array']
```

Comment:
The first one means the that the `spec` is configured for a data source column, or possibly is `null`. Are you relying on the color being `null` or `undefined` to not draw? If so I'd recommend testing the `doit` attribute to turn things on or off like the other renderers do. 

For the second I guess your usage actually differs a bit from the other renderers, `array` actually takes the data source to read from as an argument. Maybe this will have to do for now.